### PR TITLE
Cleanup Series arithmetic tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Also a few for TimedeltaIndex.  One of the TimedeltaIndex tests specifically tests a behavior I think is wrong, #18963.  Actually _fixing_ that bug is separate.

Actual content of tests is unchanged.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Travis error in TestTSPlot.test_time looks unrelated.

File: pandas/tests/indexes/timedeltas/test_arithmetic.py [hunk 1]
```diff
@@ -26,41 +26,60 @@ def freq(request):
...
+
+    def test_tdi_mul_intarray_1dim(self):
+        idx = TimedeltaIndex(np.arange(5, dtype='int64'))
 
         result = idx * np.arange(5, dtype='int64')
-        tm.assert_index_equal(result, didx)
+        expected = TimedeltaIndex(np.arange(5, dtype='int64') ** 2)
+        tm.assert_index_equal(result, expected)
+
+    # FIXME: These ops should return Series, NOT indexes
```
Comment (jreback) [inline pandas/tests/indexes/timedeltas/test_arithmetic.py]: TODO

File: pandas/tests/series/test_operators.py [hunk 2]
```diff
@@ -666,9 +666,60 @@ def test_div(self):
             result = pd.Series(zero_array) / pd.Series(data)
             assert_series_equal(result, expected)
 
+    def test_series_radd_more(self):
+        data = [[1, 2, 3],
+                [1.1, 2.2, 3.3],
+                [pd.Timestamp('2011-01-01'), pd.Timestamp('2011-01-02'),
+                 pd.NaT],
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: paramaterize

File: pandas/tests/series/test_operators.py [hunk 3]
```diff
@@ -666,9 +666,60 @@ def test_div(self):
...
+            assert_series_equal(pd.Timedelta('3 days') + s, exp)
+            assert_series_equal(s + pd.Timedelta('3 days'), exp)
+
+        s = pd.Series(['x', np.nan, 'x'])
+        assert_series_equal('a' + s, pd.Series(['ax', np.nan, 'ax']))
+        assert_series_equal(s + 'a', pd.Series(['xa', np.nan, 'xa']))
+
+    def test_invalid_add_sub(self):
+        # invalid ops
+        obj_series = tm.makeObjectSeries()
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: can you add a comment

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_operators.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: Sure.  Now that I actually look at `tm.makeObjectSeries()` I'm surprised that it gives a Series[datetime64] with what looks like random strings for its Index.  I'll move this down into TestDatetimeSeriesArithmetic

Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: Incidentally, this is one of the cases that `DatetimeIndex.__add__`/`__sub__` messes up.

**Comment by jbrockmendel** [comment]
Test error still looks unrelated and this is bound to cause merge headaches for other higher-priority PRs.  Closing for now, will circle back eventually.

**Pull request closed without merging**